### PR TITLE
Algorithm cleanup

### PR DESCRIPTION
**Pull request by milseman**

### CODE REVIEW

File: Documentation/Evolution/StringProcessingAlgorithms.md [hunk 1]
```diff
@@ -285,6 +308,8 @@ extension BidirectionalCollection where SubSequence == Substring {
 
 #### Trim prefix
 
+We propose generic `trimPrefix` for collections taking a `Element` predicate or a specific `Element`.
```
Comment (natecook1000) [inline Documentation/Evolution/StringProcessingAlgorithms.md:311]:
```suggestion
We propose generic `trimmingPrefix` and `trimPrefix` methods for collections that trim elements matching a predicate or a possible prefix sequence.
```

**Review by natecook1000 — COMMENTED** [review]
Looks good — as far as I can tell these hit all our edits. One note, and then I'll follow up later with info about empty matches.

**Comment by milseman** [comment]
@swift-ci please test

**Review by itingliu — APPROVED** [review]
Looks good. I like separating regexcomponent-backed ones from the collection ones. Also I'm glad that we can land on `Collection` instead of `BidirectionalCollection`

**Pull request merged**